presto-spi/src/main/java/com/facebook/presto/spi/SourceLocation.java

```diff
@@ -47,4 +49,30 @@ public int getColumn()
     {
         return column;
     }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+
+        SourceLocation that = (SourceLocation) obj;
+
+        return line == that.line && column == that.column;
```

Comment:
you can use Objects.equals

Reply:
Already done

Reply:
I think you used Objects.hash for the hashCode. This is similarly for equals.

Reply:
Line and column are int so using object equals will cause boxing. For hash code logic to mix them up is hard. So we can use objects hash code.